python/ray/_private/ray_constants.py

```diff
@@ -69,8 +69,10 @@ def env_bool(key, default):
 RAY_RUNTIME_ENV_URI_PIN_EXPIRATION_S_ENV_VAR = (
     "RAY_RUNTIME_ENV_TEMPORARY_REFERENCE_EXPIRATION_S"
 )
-# Defaults to 30 seconds. This should be enough time for the job to start.
-RAY_RUNTIME_ENV_URI_PIN_EXPIRATION_S_DEFAULT = 30
+# Defaults to 10 minutes. This should be longer than the total time it takes for
+# the local working_dir and py_modules to be uploaded, or these files might get
+# garbage collected before the job starts.
+RAY_RUNTIME_ENV_URI_PIN_EXPIRATION_S_DEFAULT = 10 * 60
```

Comment:
why do we have expiration in the first place? Isn't it possible to just ref count?

Reply:
There's more discussion about this in https://github.com/ray-project/ray/pull/24719.  If we don't have a timeout for this temporary reference somewhere when we upload the package from the Job CLI or Ray Client, if the user ctrl-C's from their laptop before `ray.init()` starts on the cluster, the temporary reference would never be decremented.

@wuisawesome I think you had a good explanation about this, do you mind sharing it here? It was some general principle about how to keep track of resource ownership in a distributed system, you either need a lease or some kind of session with a heartbeat, and here we've opted for a lease.

Reply:
> why do we have expiration in the first place? Isn't it possible to just ref count?

This is being uploaded by a user laptop, and there's no heartbeat to know if the laptop is alive, so if we were to ref count it and the laptop dies, the runtime env would be leaked (a potential problem since runtime envs are actually quite large in practice)